app/routes.py

```diff
@@ -1,2 +1,193 @@
-from flask import Blueprint
+from app import db
+from flask import Blueprint, request, abort, jsonify, make_response
+from datetime import datetime
+from dotenv import load_dotenv
+import os
+import requests
+from app.models.task import Task
+from app.models.goal import Goal
+from app.utils.route_wrappers import require_instance_or_404
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    """
+    Retrieve all tasks. Allows for use of query parameters.
+    Returns JSON list of task dictionaries. """
+    query = Task.query # Base query
+
+    # Query params, adding to query where indicated
+    sort = request.args.get("sort")
+    if sort == "asc":
+        query = query.order_by(Task.title)
+    elif sort == "desc":
+        query = query.order_by(Task.title.desc())
+    
+    query = query.all() # Final query
+    
+    # Returns jsonified list of task dicionaries
+    return jsonify([task.to_dict() for task in query]), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+@require_instance_or_404
+def get_task(task):
+    """Retrieve one stored task by id."""
+    if task.goal_id:
+        return jsonify({"task": task.goals_to_dict()}), 200
+    else:
+        return jsonify({"task": task.to_dict()}), 200
+
+@tasks_bp.route("", methods=["POST"])
+def post_task():
+    """Create a new task from JSON data."""
+    form_data = request.get_json()
+
+    # All fields must be provided
+    mandatory_fields = ["title", "description", "completed_at"]
+    for field in mandatory_fields:
+        if field not in form_data:
+            return jsonify({"details": "Invalid data"}), 400 
+
+    new_task = Task(
+        title=form_data["title"],
+        description=form_data["description"],
+        completed_at=form_data["completed_at"]
+    )
+
+    db.session.add(new_task)
+    db.session.commit()
+    return {"task": new_task.to_dict()}, 201
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+@require_instance_or_404
```

Comment:
Nice work with this decorator!  Setting up `task` as the parameter in the function makes the code in the function very streamlined!